### PR TITLE
Deployment selector should be immutable

### PR DESCRIPTION
**Pull request by polyrabbit**
The checklist before PR is ready for review:

- [ ] has unit testing for new added codes
- [ ] has functional testing for new added features
- [x] has checked the lint or style issues
- [ ] README updated if need

Bind the deployment selector to a specified fx app, which should be unique in the whole cluster.

Deployment selector is immutable over the life cycle of a deployment. The current implementation generates a new UUID even when user tries to **update** a deployment, which will be rejected by k8s as:

```bash
fatal deploy function hello (func.go) failed: Deployment.apps "hello" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"fx-app-cf65f89a-cfd1-4a1f-8f8e-ce74fd187488"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

### CODE REVIEW

File: deploy/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -43,6 +35,7 @@ func (k *K8S) Deploy(
 	name string,
 	ports []int32,
 ) error {
+	// TODO: should use the one specified in config file
```
Comment (metrue) [inline deploy/kubernetes/kubernetes.go]: tracked by https://github.com/metrue/fx/issues/318

File: container_runtimes/docker/sdk/docker.go [hunk 2]
```diff
@@ -69,10 +69,9 @@ func (d *Docker) BuildImage(ctx context.Context, workdir string, name string) er
 	if err != nil {
 		return err
 	}
+	defer resp.Body.Close()
 
 	if os.Getenv("DEBUG") != "" {
-		defer resp.Body.Close()
-
```
Comment (metrue) [inline container_runtimes/docker/sdk/docker.go]: Great fix.

File: handlers/deploy.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func Deploy(cfg config.Configer) HandleFunc {
 			}
 
 			if err != nil {
-				log.Fatalf("deploy function %s (%s) failed: %v", err)
+				log.Fatalf("deploy function %s (%s) failed: %v", name, funcFile, err)
```
Comment (metrue) [inline handlers/deploy.go]: Great catch.

File: deploy/kubernetes/kubernetes.go [hunk 4]
```diff
@@ -61,7 +53,7 @@ func (k *K8S) Deploy(
 	// be created automatically, then incoming traffic to Service will be forward to Pod.
 	// Then we have no need to create Endpoint manually anymore.
 	selector := map[string]string{
-		"app": "fx-app-" + uuid.New().String(),
+		"app": "fx-app-" + name,
```
Comment (pplam) [inline deploy/kubernetes/kubernetes.go]: Maybe we should check if a name is used here

Comment (polyrabbit) [reply deploy/kubernetes/kubernetes.go]: Deployment name is guaranteed to be unique by [Line 60](https://github.com/polyrabbit/fx/blob/f92491d7e0a9d22df21f238cc619add9eccd3d67/deploy/kubernetes/kubernetes.go#L60). So I presume the name with a `"fx-app-"` prefix should be unique too.

To be precise, Yes, we need more check here.

**Pull request merged**